src/test_linked_list.py

```diff
@@ -0,0 +1,100 @@
+# coding=utf-8
+from __future__ import unicode_literals, print_function
+from builtins import *
+
+import pytest
+
+
+CONSTRUCTOR_LISTS = [
+    ([], []),
+    ([1, 2, 3], [3, 2, 1]),
+    ([5], [5]),
+    ([3, 4, [3, 4, 5], 3, None], [None, 3, [3, 4, 5], 4, 3]),
+    ("asdf", ['f', 'd', 's', 'a']),
+    ([1, 1, 1, 1, 1], [1, 1, 1, 1, 1]),
+]
+
+
+@pytest.mark.parametrize('values, result', CONSTRUCTOR_LISTS)
+def test_constructor(values, result):
+    from linked_list import LinkedList
+    assert list(LinkedList(values)) == result
+
+
+def test_constructor_failure():
+    from linked_list import LinkedList
+    # None is not iterable
+    with pytest.raises(TypeError):
+        LinkedList(None)
+
+
+def test_empty_list():
+    from linked_list import LinkedList
+    a_list = LinkedList()
+    assert a_list.length == 0
+    assert list(a_list) == []
+
+
+def test_insert():
```

Comment:
Typically you don't want more than one or two asserts per test. If the last two pass, the first two must pass. I get what you're doing, and it makes sense.. This is more of a convention comment.